avm/res/devops-infrastructure/managed-devops-pools/main.bicep

```diff
@@ -0,0 +1,448 @@
+}
+
+@discriminator('kind')
+type organizationProfileType = organizationGitHubType | organizationAzureDevOpsType
+
+type secretManagementType = {
+  @description('Required. The secret management settings of the machines in the pool.')
+  keyExportable: bool
+
+  @description('Required. The list of certificates to install on all machines in the pool.')
+  observedCertificates: string[]
+
+  @description('Optional. Where to store certificates on the machine.')
+  certificateStoreLocation: string?
+}?
+
+type dataDiskType = {
+  @description('Optional. The type of caching to be enabled for the data disks. The default value for caching is readwrite. For information about the caching options see: https://blogs.msdn.microsoft.com/windowsazurestorage/2012/06/27/exploring-windows-azure-drives-disks-and-images/.')
+  caching: ('None' | 'ReadOnly' | 'ReadWrite')?
+
+  @description('Optional. The initial disk size in gigabytes.')
+  diskSizeGiB: int?
+
+  @description('Optional. The drive letter for the empty data disk. If not specified, it will be the first available letter. Letters A, C, D, and E are not allowed.')
+  driveLetter: string?
+
+  @description('Optional. The storage Account type to be used for the data disk. If omitted, the default is Standard_LRS.')
+  storageAccountType: ('Premium_LRS' | 'Premium_ZRS' | 'StandardSSD_LRS' | 'StandardSSD_ZRS' | 'Standard_LRS')?
+}[]?
+
+type resourcePredictionsProfileAutomaticType = {
+  @description('Required. The stand-by agent scheme is determined based on historical demand.')
+  kind: 'Automatic'
+
+  @description('Required. Determines the balance between cost and performance.')
+  predictionPreference: 'Balanced' | 'MostCostEffective' | 'MoreCostEffective' | 'MorePerformance' | 'BestPerformance'
+}
+
+type resourcePredictionsProfileManualType = {
+  @description('Required. Customer provides the stand-by agent scheme.')
+  kind: 'Manual'
+}
+
+type agentStatefulType = {
+  @description('Required. Stateful profile meaning that the machines will be returned to the pool after running a job.')
+  kind: 'Stateful'
+
+  @description('Required. How long should stateful machines be kept around. The maximum is one week.')
+  maxAgentLifetime: string
+
+  @description('Required. How long should the machine be kept around after it ran a workload when there are no stand-by agents. The maximum is one week.')
+  gracePeriodTimeSpan: string
+
+  @description('Optional. Defines pool buffer/stand-by agents.')
+  resourcePredictions: object?
+
+  @discriminator('kind')
+  @description('Optional. Determines how the stand-by scheme should be provided.')
+  resourcePredictionsProfile: (resourcePredictionsProfileAutomaticType | resourcePredictionsProfileManualType)?
+}
+
+type agentStatelessType = {
+  @description('Required. Stateless profile meaning that the machines will be cleaned up after running a job.')
+  kind: 'Stateless'
+
+  @description('Optional. Defines pool buffer/stand-by agents.')
+  resourcePredictions: {
```

Comment:
This and the `daysData` property are missing metadata. "Should've" been picked up by test

Reply:
Did not run the test yet, but added the required decorators! 😄